src/wasm/wasm-stack.cpp

```diff
@@ -1940,21 +1940,38 @@ void BinaryInstWriter::visitStructSet(StructSet* curr) {
 }
 
 void BinaryInstWriter::visitArrayNew(ArrayNew* curr) {
-  WASM_UNREACHABLE("TODO (gc): array.new");
+  o << int8_t(BinaryConsts::GCPrefix);
+  if (curr->isWithDefault()) {
+    o << U32LEB(BinaryConsts::ArrayNewDefaultWithRtt);
+  } else {
+    o << U32LEB(BinaryConsts::ArrayNewWithRtt);
+  }
+  parent.writeHeapType(curr->rtt->type.getHeapType());
 }
 
 void BinaryInstWriter::visitArrayGet(ArrayGet* curr) {
-  WASM_UNREACHABLE("TODO (gc): array.get");
+  const auto& heapType = curr->ref->type.getHeapType();
```

Comment:
```suggestion
  auto heapType = curr->ref->type.getHeapType();
```
Huh, TIL you can have a const reference to an rvalue. Not taking a reference to the HeapType should be more idiomatic, though.